modules/dictionary.lobster

```diff
@@ -35,15 +36,37 @@ class dictionary<K,V>:
         if c:
             c.value = value
         else:
+            sz++
             buckets[h] = chain<K,V> { buckets[h], key, value }
 
+    def del(key:K):
+        let h = hash(key) % buckets.length
+        var c = buckets[h]
+        var prev = nil::chain<K,V>
+        while c:
```

Comment:
this algorithm is a bit lengthy and clumsy, probably would be neater as:
```
def delete(c):
    if c.key == key:
        return c.next
    c.next = delete(c.next)
    return c
buckets[h] = delete(buckets[h)
```

Reply:
It is neater, at the cost of more overhead with recursion calls. Though hash collision should be considered rare case I guess? Will recommit later